openpype/hosts/maya/plugins/publish/extract_rig_fbx.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+import os
+
+from maya import cmds  # noqa
+import maya.mel as mel  # noqa
+import pyblish.api
+
+from openpype.pipeline import publish
+from openpype.hosts.maya.api.lib import maintained_selection
+from openpype.hosts.maya.api import fbx
+
+
+class ExtractRigFBX(publish.Extractor):
+    """Extract Rig in FBX format from Maya.
+
+    This extracts the rig in fbx with the constraints
+    and referenced asset content included.
+    This also optionally extract animated rig in fbx with
+    geometries included.
+
+    """
+    order = pyblish.api.ExtractorOrder
+    label = "Extract Rig (FBX)"
+    families = ["rig"]
+
+    def process(self, instance):
+        if not instance.data.get("fbx_enabled"):
+            self.log.debug("fbx extractor has been disable.."
+                           "Skipping the action...")
+            return
+
+        # Define output path
+        staging_dir = self.staging_dir(instance)
+        filename = "{0}.fbx".format(instance.name)
+        path = os.path.join(staging_dir, filename)
+
+        # The export requires forward slashes because we need
+        # to format it into a string in a mel expression
+        path = path.replace('\\', '/')
+
+        self.log.debug("Extracting FBX to: {0}".format(path))
+
+        control_rigs = instance.data.get("control_rigs", [])
+        skeletal_mesh = instance.data.get("skeleton_mesh", [])
+        members = control_rigs + skeletal_mesh
+        self._to_extract(instance, path, members)
+
+        if "representations" not in instance.data:
+            instance.data["representations"] = []
+
+        representation = {
+            'name': 'fbx',
+            'ext': 'fbx',
+            'files': filename,
+            "stagingDir": staging_dir,
+        }
+        instance.data["representations"].append(representation)
+
+        self.log.debug("Extract FBX successful to: {0}".format(path))
+        if skeletal_mesh:
+            self._to_extract(instance, path, skeletal_mesh)
+            representation = {
+                'name': 'fbxanim',
```

Comment:
Why can't this just be `fbx`? Is there another fbx representation we're expecting for a rig. And when does a `rig` publish actually contain animation - that seems wrong. `animation` family publishes what is the resulting animation made with a loaded rig.